Tests/FoundationEssentialsTests/FileManager/FileManagerTests.swift

```diff
@@ -458,11 +458,7 @@ final class FileManagerTests : XCTestCase {
             XCTAssertEqual(try $0.subpathsOfDirectory(atPath: ".").sorted(), ["dir", "dir/bar", "dir/foo", "dir2", "dir2/bar", "dir2/foo", "other_file"])
             XCTAssertEqual($0.contents(atPath: "dir/foo"), data)
             XCTAssertEqual($0.contents(atPath: "dir2/foo"), data)
-#if os(Windows)
```

Comment:
Can you elaborate on how the change to this API affects the behavior here? I wouldn't have expected this to change the order in which the directory is iterated

Reply:
Dropped the change. I somehow had convinced myself that it was due _FileManagerImpl.createDirectory calling String.deletingLastPathComponent for the temporary paths. When in fact the unexpected local test failure is completely unrelated to my local change. e.g dropping my change

```
$ uname -a
Linux ThinkStation-06db 5.15.0-107-generic #117-Ubuntu SMP Fri Apr 26 12:26:49 UTC 2024 x86_64 x86_64 x86_64 GNU/Linux
$ cat /etc/lsb-release 
DISTRIB_ID=Ubuntu
DISTRIB_RELEASE=22.04
DISTRIB_CODENAME=jammy
DISTRIB_DESCRIPTION="Ubuntu 22.04 LTS"
$ git reset --hard ef8c1d539cd8df3da5151befd95afee5fa64890e
HEAD is now at ef8c1d5 Set SWIFT_SYSTEM_NAME for macro project in CMake (#896)
$ swift  test --filter testCopyItemAtPathToPath
Building for debugging...
[1/1] Write swift-version-2B7046DC1514D7B6.txt
Build complete! (0.57s)
Test Suite 'Selected tests' started at 2024-09-05 14:44:10.027
Test Suite 'FileManagerTests' started at 2024-09-05 14:44:10.030
Test Case 'FileManagerTests.testCopyItemAtPathToPath' started at 2024-09-05 14:44:10.030
/home/ryan_mansfield/swift/swift-foundation/Tests/FoundationEssentialsTests/FileManager/FileManagerTests.swift:464: error: FileManagerTests.testCopyItemAtPathToPath : XCTAssertEqual failed: ("['dir' --> 'dir2', 'dir/bar' --> 'dir2/bar', 'dir/foo' --> 'dir2/foo']") is not equal to ("['dir' --> 'dir2', 'dir/foo' --> 'dir2/foo', 'dir/bar' --> 'dir2/bar']") - 
Test Case 'FileManagerTests.testCopyItemAtPathToPath' failed (0.039 seconds)
Test Suite 'FileManagerTests' failed at 2024-09-05 14:44:10.069
	 Executed 1 test, with 1 failure (0 unexpected) in 0.039 (0.039) seconds
Test Suite 'Selected tests' failed at 2024-09-05 14:44:10.069
	 Executed 1 test, with 1 failure (0 unexpected) in 0.039 (0.039) seconds
◇ Test run started.
↳ Testing Library Version: unknown
✔ Test run with 0 tests passed after 0.001 seconds.

```